### DISCUSSION TITLE
Question about a.any() and a.all() in the _evaluate method

### DISCUSSION THREAD

**Discussion by Atulsingh92** [anyoptimization/pymoo#206] [Q&A] [answered]
Hello, 

I have a following piece of code within my very long _evaluate method , which has many formulas, one of which is as follows, 

```
plast = 10
while np.abs(Dpf-plast)>1:
    plast = Dpf
    Dpf = all(143.5 * Nr * zeta * pow(Ga,2) * (273+thm) / (P*1e5 - Dpf/2)) #finds Dpf iteratively
```

here only thm, P are a single valued floats, whereas, others, i.e Nr, zeta, Ga, and Dpf eventually are all 
``` (population_size, ) ``` arrays.

My question is if I run a separate file of my all the formulas that I have now kept inside the _evaluate method, I do not require to give `all()` or `any()` in that file, however inside the _evaluate method with those formulas, I get an error to which said use either of the two. Hence you see the `all()` in the above while loop.

Would this affect the result in anyway, or rather is there a way I could run all the formulas of the _evaluate method, like I would have run in a separate file, i.e. without using `any()` or `all()`?

**Comment by blankjul** [comment, marked as answer]
pymoo does not really care how you obtain the `F` and `G` values, as long as you set them correctly to the `out` array.
You can even call third-party software, run your evaluation on a cluster, and so on. As long as you set them to `out` when you have obtained all values.
